skynet.go

```diff
@@ -16,21 +16,67 @@ import (
 )
 
 type (
+	// Skykey contains information about a skykey.
+	Skykey struct {
+		Skykey string `json:"skykey"`
+		Name   string `json:"name"`
+		ID     string `json:"id"`
+		Type   string `json:"type"`
+	}
+
+	// UploadData contains data to upload, indexed by filenames.
+	UploadData map[string]io.Reader
+
+	// Options structs.
```

Comment:
nit: Something we do in various places in the Sia codebase that might be helpful here is doing multiple `type` blocks especially If we are planning on keeping all this in one file.

So something like the following
```go
// General Options
type (
   ConnectionOptions
  ...
)

// SkyKey Options
type (
  AddSkykeyOptions
  CreateSkykeyOptions
  ...
)
```
That helps group things and justifies alphabetizing sections and not the entire list. 

If you agree we could lump that change into the next SDK update when I'm sure there will be more option structs added.

Reply:
I actually want to create separate files for each feature e.g. `upload.go`, `download.go`, `encryption.go`. etc. I already started doing this in Python (we already have separate files in node-js). With the increasing amount of code this will significantly improve organization

Reply:
ok perfect, that is better. Wasn't sure if there was a reason we were keeping everything in one file.